lintro/tools/definitions/tsc.py

```
@@ -541,179 +530,329 @@ def check(self, paths: list[str], options: dict[str, object]) -> ToolResult:
 
-                # Generic failure
-                return ToolResult(
-                    name=self.definition.name,
-                    success=False,
-                    output=normalized_output or "tsc execution failed.",
-                    issues_count=0,
+    # -----------------------------------------------------------------
+    # Shared tsc execution + output parsing
+    # -----------------------------------------------------------------
+
+    def _run_tsc_and_parse(
```

Ruff (0.15.9)

[warning] Too many return statements (8 > 6)

(PLR0911)